solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/LTRQParserPlugin.java

```diff
@@ -78,13 +78,23 @@ public Query parse() throws SyntaxError {
         throw new SolrException(SolrException.ErrorCode.BAD_REQUEST,
             "Must provide model in the request");
       }
-
+      
+      
+      
+     
       final LTRScoringAlgorithm meta = mr.getModel(modelName);
       if (meta == null) {
         throw new SolrException(ErrorCode.BAD_REQUEST,
             "cannot find " + CommonLTRParams.MODEL + " " + modelName);
       }
-      final ModelQuery reRankModel = new ModelQuery(meta);
+      final String modelFeatureStoreName = meta.getFeatureStoreName();
+      final Boolean extractFeatures = (Boolean) req.getContext().get(CommonLTRParams.LOG_FEATURES_QUERY_PARAM);
+      final String fvStoreName = (String) req.getContext().get(CommonLTRParams.STORE);
+      boolean featuresRequested = (extractFeatures != null && (modelFeatureStoreName == fvStoreName || fvStoreName == null) ) ? extractFeatures.booleanValue():false;
```

Comment:
This condition and the fvCache condition below look very similar.  Can we combine them together to make both of them a little bit more readable?